estudios_socioeconomicos/serializers.py

```diff
@@ -63,3 +69,117 @@ class Meta:
             'nombre',
             'numero',
             'subsecciones')
+
+
+class RespuestaSerializer(serializers.ModelSerializer):
+    """ Serializer for using .models.Respuesta objects
+        in REST endpoint.
+    """
+    class Meta:
+        model = Respuesta
+        fields = (
+            'estudio',
+            'pregunta',
+            'eleccion',
+            'respuesta')
+
+        read_only_fields = ('estudio', )
+
+
+class EstudioSerializer(serializers.ModelSerializer):
+    """ Serializer to represent a .models.Estudio instance
+        through a REST endpoint for the offline application
+        to submit information.
+    """
+    respuesta_estudio = RespuestaSerializer(many=True, allow_null=True)
+    familia = FamiliaSerializer()
+
+    class Meta:
+        model = Estudio
+        fields = (
+            'id',
+            'familia',
+            'status',
+            'capturista',
+            'respuesta_estudio')
+
+        read_only_fields = ('id', 'status', 'capturista')
+
+    @atomic
+    def create(self, capturista):
+        """ This function overides the default behaviour for creating
+            an object through a serializer.
+
+            Django Rest Framework does not implement saving nested
+            serializers.
+
+            Estudio object depends on a Familia insance. This function
+            first creates the familia, then creates the Estudio and
+            finally creates all answers that depend on the Estudio.
+
+            Notes
+            -----
+            Saves the Estudio instance using .save_base(raw=True). This
+            is required because Estudio has a Trigger required for online
+            client to create empy answers for any new study. The offline
+            client does not require this functionality.
+
+            save_foreign_relationship returns a list of created or updated
+            objects. We just care about the first.
+
+            The familia_data.pop('id', None) is just a defense mechanism in
+            case somebody for some strange reason decides it's a good idea to
+            send a study without an ID but a family with ID. We would just
+            create e new family (family and estudio can only have 1 - 1 relashionship).
+        """
+        familia_data = self.validated_data.pop('familia')
+        respuestas = self.validated_data.pop('respuesta_estudio')
+
+        familia_data.pop('id', None)  # In the catastrofic case someone decides on sending
+
+        family_instance = save_foreign_relationship([familia_data], FamiliaSerializer, Familia)
+
+        if len(family_instance) != 1 or family_instance[0] is None:
```

Comment:
shouldn't this be before the previous call to save_foreign_relationship? or havent I understood smth?